python/src/aica_api/client.py

```diff
@@ -54,7 +59,56 @@ def _endpoint(self, endpoint=''):
         """
         if self._protocol is None:
             self.protocol()
-        return f'{self._address}/{self._protocol}/{endpoint}'
+        return self.__raw_endpoint(f'{self._protocol}/{endpoint}')
+
+    def __raw_endpoint(self, endpoint: str) -> str:
+        return f'{self._address}/{endpoint}'
+
+    def __create_token(self) -> None:
+        """Authenticate with the API and store the result in self.__token."""
+        if self.__token is not None:
+            return
+        rep = requests.post(self._endpoint('auth/login'), headers={'Authorization': f'Bearer {self.__api_key}'})
+        rep.raise_for_status()
+        self.__token = rep.json()['token']
```

Comment:
nit: `res` for response rather than `rep`?
```suggestion
        res = requests.post(self._endpoint('auth/login'), headers={'Authorization': f'Bearer {self.__api_key}'})
        res.raise_for_status()
        self.__token = res.json()['token']
```